editor/node_3d_editor_gizmos.cpp

```diff
@@ -3613,6 +3624,21 @@ void CollisionShape3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
 		Vector<Vector3> handles;
 		handles.push_back(Vector3(r, 0, 0));
 		p_gizmo->add_handles(handles, handles_material);
+
+		// Visible Faces
+		if (EDITOR_DEF_RST("editors/3d_gizmos/collision_shape/show_faces", true)) {
+			Ref<ArrayMesh> m = sp->get_debug_arraymesh_faces();
+			if (EDITOR_DEF_RST("editors/3d_gizmos/collision_shape/random_color", true)) {
+				Ref<StandardMaterial3D> shape_face_color_mat = memnew(StandardMaterial3D);
+				const Color shape_face_color = Color().from_hsv(hash_one_uint64(sp->get_instance_id()) / (float)UINT32_MAX, 1.0, 1.0, 1.0);
+				shape_face_color_mat->set_albedo(Color(shape_face_color, cs->is_disabled() ? 0.25 : 0.66));
+				shape_face_color_mat->set_transparency(StandardMaterial3D::TRANSPARENCY_ALPHA);
+				shape_face_color_mat->set_shading_mode(StandardMaterial3D::SHADING_MODE_UNSHADED);
+				m->surface_set_material(0, shape_face_color_mat);
+			} else
+				m->surface_set_material(0, get_material(cs->is_disabled() ? "face_material_disabled" : "face_material"));
+			p_gizmo->add_mesh(m);
+		}
 	}
```

Comment:
If I'm not mistaken, this code is repeated for every collision shape type. Can we have it just once at the top of the function and store the material in a variable that is then used on the various `add_mesh()` calls?